gconf/handler.go

```diff
@@ -20,17 +20,42 @@ type OwnedConfig interface {
 type UpdateConfigurationHandler struct {
 	pkg string
 	// We require this type to load the data.
-	config OwnedConfig
-	auth   x.Authenticator
+	config    OwnedConfig
+	auth      x.Authenticator
+	initAdmin func(weave.ReadOnlyKVStore) (weave.Address, error)
 }
 
 var _ weave.Handler = (*UpdateConfigurationHandler)(nil)
 
-func NewUpdateConfigurationHandler(pkg string, config OwnedConfig, auth x.Authenticator) UpdateConfigurationHandler {
+// NewUpdateConfigurationHandler returns a message handler that process
+// configuration patch message.
+//
+// To pass authentication step, each message must be signed by the current
+// configuration owner.
+//
+// A special chicken-egg problem appears when the configuration does not exist
+// (it was not created via genesis). This is an issue, because without
+// configuration we cannot configure configuration owner that can update the
+// configuration. This means that the configuration cannot be created as well.
+// A configuration is needed to create a configuration.
+// To address the above issue, an optional `initConfAdmin` argument can be
+// given to provide a creation only admin address. A good deafult is to use
+// `migration.CurrentAdmin` function.
+// `initConfAdmin` is used to authenticate the tranaction only when no
+// configuration exist. Once a configuration is created, `initConfAdmin` is not
+// used anymore and the autentication relies only on configuration's owner
+// declaration.
+func NewUpdateConfigurationHandler(
+	pkg string,
+	config OwnedConfig,
+	auth x.Authenticator,
+	initConfAdmin func(weave.ReadOnlyKVStore) (weave.Address, error),
```

Comment:
@ethanfrey @alpe Please have a look if you have a moment. I think this is a rather big change to `weave` configuration patterns.

This is solving the problem I have described in the comment above - I want to create a configuration and cannot authenticate, because configuration entity defines who can authenticate :hatching_chick: 

Usually configuration is created via genesis. This is not possible if you add an extension to an existing state.

It is not possible to do this via migrations as Ethan suggested for two reasons:
- migrations do not have access to the context. This is on purpose as the code that runs them might not have access to the context as well (i.e. bucket),
- migrations are given only `weave.ReadOnlyKVStore`. This is also on purpose, so that they are not able to modify the database state (!). They are supposed to migrate the in memory entity representation.

I require explicit passing of an admin address function instead of directly importing `migration.loadConf` in `gconf` because of import cycle. I think it is also nice that `initConfAdmin` is optional.

What do you guys think?